tests/conftest.py

```diff
@@ -192,7 +193,9 @@ async def connector(fake_credentials: Credentials) -> AsyncGenerator[Connector,
         mock_auth.return_value = fake_credentials, None
         # mock Cloud SQL Admin API calls
         mock_instance = FakeCSQLInstance(project, region, instance_name)
-        _, client_key = connector._keys.result()
+        _, client_key = asyncio.run_coroutine_threadsafe(
+            wait_for_keys(connector._keys), connector._loop
+        ).result()
```

Comment:
@kurtisvg here is the reason for `wait_for_keys` helper function. 

In the `connector` fixture that creates a `Connector` and mocks the API calls, I need the client key in order to mock the ephemeral cert. Since we now call `wrap_future(keys)` in the Connector's __init__ I can no longer call `connector._keys.result()` since the keys are now an `asyncio.Future` instead of `concurrent.Future`. A concurrent future will block on `.result()` and return the result once it is finished (does not need to be awaited), however `.result()` on an asyncio future will throw an error if result is not set when it is called (needs to be await prior to calling .result). This helper function basically allows me to turn an `asyncio.Future` into a `concurrent.Future`, unwrapping the future so I can access its value and mock the API calls prior to creating an `Instance` object.

Reply:
I'm not sure I follow. Why don't you just await the keys (like it's done in the test on L160)?

Even if we do need this, it seems like it should be an anonymous function here as opposed to something in mocks.

Reply:
I cleaned up L160 to just use a task, it works because the present event loop is running in the current thread and can be awaited. This is not the case because for the `connector` fixture. The `connector._loop` is running in the connector's background thread and can not be awaited from the fixture, thus the need to have the helper function await in the background thread so the result can be accessed within the fixture.

Reply:
Ok so to summarize, you are converting an awaitable (asyncio.Future) into a concurrent.Future. You are using a closure to convert your awaitable into a coro because `run_coroutine_threadsafe` only takes coros.

It is probably worth trying to leave a comment that tries to summarize that. 

Also, make sure you've ruled out using `asyncio.wait()` -- it looks like it returns a coro, so it might be used here.

Reply:
Added comments and ruled out `asyncio.wait()` by testing it.